### PR TITLE
Fix flake8 violations

### PR DESCRIPTION
**Pull request by joaander**
## Description

Change Python code to deal with all flake8 violations. In some cases, add ignores or remove unused files.

## Motivation and context

Improve code style so we can enforce pre-commit going forward.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
I would like to merge this on Friday so that we can have the style checks in place for the hackathon.

No pressure on the review, any documentation or other issues not caught by the unit tests can always be fixed at a later time.

**Comment by joaander** [comment]
I'm working on fixing the sphinx doc formatting.

File: hoomd/conftest.py [hunk 1]
```diff
@@ -61,20 +67,21 @@ def make_simulation(snapshot=None):
...
+
+        Args:
+            particle_types: List of particle type names
+            dimensions: Number of dimensions (2 or 3)
+            d: Distance apart to place particles
+            L: Box length
+
+        The two particles are placed at (-d/2, 0, 0) and (d/2,0,0). When,
+        dimensions==3, the box is L by L by L. When dimensions==2, the box is
+        L by L by 1.
```
Comment (b-butler) [inline hoomd/conftest.py:83]: This is wrong the third dimension is 0.

File: hoomd/data/syncedlist.py [hunk 2]
```diff
@@ -32,6 +38,7 @@ def __call__(self, obj):
 
 
 def identity(obj):
+    """Returns x."""
```
Comment (b-butler) [inline hoomd/data/syncedlist.py:41]:
```suggestion
    """Returns obj."""
```

**Pull request merged**